src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CcdCaseUpdater.java

```diff
@@ -0,0 +1,205 @@
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.servicebus.domains.payments.PaymentsPublishingException;
+import uk.gov.hmcts.reform.ccd.client.CoreCaseDataApi;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDataContent;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.ccd.client.model.Event;
+import uk.gov.hmcts.reform.ccd.client.model.StartEventResponse;
+
+import java.util.Map;
+
+import static java.lang.String.format;
+
+@Service
+public class CcdCaseUpdater {
+    private static final Logger log = LoggerFactory.getLogger(CcdCaseUpdater.class);
+
+    private static final String EXCEPTION_RECORD_REFERENCE = "bulkScanCaseReference";
+
+    private final CaseUpdateClient caseUpdateClient;
+    private final AuthTokenGenerator s2sTokenGenerator;
+    private final CoreCaseDataApi coreCaseDataApi;
+    private final ExceptionRecordFinalizer exceptionRecordFinalizer;
+
+    public CcdCaseUpdater(
+        CaseUpdateClient caseUpdateClient,
+        AuthTokenGenerator s2sTokenGenerator,
+        CoreCaseDataApi coreCaseDataApi,
+        ExceptionRecordFinalizer exceptionRecordFinalizer
+    ) {
+        this.caseUpdateClient = caseUpdateClient;
+        this.s2sTokenGenerator = s2sTokenGenerator;
+        this.coreCaseDataApi = coreCaseDataApi;
+        this.exceptionRecordFinalizer = exceptionRecordFinalizer;
+    }
+
+    @SuppressWarnings({"squid:S2139", "unchecked"}) // squid for exception handle + logging
+    public void updateCase(
+        ExceptionRecord exceptionRecord,
+        ServiceConfigItem configItem,
+        String idamToken,
+        String userId,
+        CaseDetails existingCase
+    ) {
+        log.info(
+            "Start updating case for {} with case ID {} from exception record {}",
+            configItem.getService(),
+            existingCase.getId(),
+            exceptionRecord.id
+        );
+
+        try {
+            String s2sToken = s2sTokenGenerator.generate();
+
+            SuccessfulUpdateResponse updateResponse = caseUpdateClient.updateCase(
+                configItem.getUpdateUrl(),
+                existingCase,
+                exceptionRecord,
+                s2sToken
+            );
+
+            log.info(
+                "Successfully updated case for {} with case ID {} from exception record {}",
+                configItem.getService(),
+                existingCase.getId(),
+                exceptionRecord.id
+            );
+
+            checkBulkScanReferenceIsSet(
+                (Map<String, ?>) updateResponse.caseDetails.caseData,
+                exceptionRecord.id
+            );
+
+            updateCaseInCcd(
+                idamToken,
+                s2sToken,
+                userId,
+                exceptionRecord,
+                updateResponse.caseDetails,
+                existingCase
+            );
+
+            log.info(
+                "Successfully updated case for {} with case ID {} from exception record {}",
```

Comment:
Worth including word `service` in the message. It was supposed to be done for case creation, as well, but that change is missing.

Reply:
Will do